### PR TITLE
Add unittest

### PR DESCRIPTION
**Pull request by kennymckormick**

### CODE REVIEW

**Comment by innerlee** [comment]
The two mp4 files are too large, could you reduce their sizes?

image

File: tests/test_data/test_dataset.py [hunk 1]
```diff
@@ -142,8 +149,34 @@ def setup_class(cls):
...
 
+    def test_rawvideo_dataset(self):
+        # Try to load txt file
+        rawvideo_dataset = RawVideoDataset(
+            ann_file=self.rawvideo_test_anno_txt,
+            pipeline=self.rawvideo_pipeline,
+            clipname_tmpl='part_{}.mp4',
+            sampling_strategy='positive',
+            data_prefix=self.data_prefix)
+        result = rawvideo_dataset.prepare_train_frames(0)
```
Comment (dreamerlin) [inline tests/test_data/test_dataset.py:166]:
```python
result = rawvideo_dataset[0]
```

File: tests/test_data/test_dataset.py [hunk 2]
```diff
@@ -142,8 +149,34 @@ def setup_class(cls):
...
+        assert result['filename'] == clipname
+
+        # Try to load json file
+        rawvideo_dataset = RawVideoDataset(
+            ann_file=self.rawvideo_test_anno_json,
+            pipeline=self.rawvideo_pipeline,
+            clipname_tmpl='part_{}.mp4',
+            sampling_strategy='random',
+            data_prefix=self.data_prefix)
+        result = rawvideo_dataset.prepare_test_frames(0)
```
Comment (dreamerlin) [inline tests/test_data/test_dataset.py:178]: add `test_mode=True` and `result = rawvideo_dataset[0]`, unittest should be perform in the same way with the normal calling.

File: tests/test_data/test_dataset.py [hunk 3]
```diff
@@ -193,6 +226,28 @@ def test_hvu_dataset(self):
...
+            ]),
+            np.array([
+                0.87904563, 1.64264224, 0.46382051, 0.72865088, -2.13712525,
+                1.28571358, 1.01320328, 0.59292737, -0.05502892
+            ])
+        ]
+        mAP = hvu_video_eval_dataset.evaluate(results)
+        assert np.isclose(mAP['action_mAP'], 1.0)
+        assert np.isclose(mAP['scene_mAP'], 0.5)
+        assert np.isclose(mAP['object_mAP'], 0.75)
```
Comment (dreamerlin) [inline tests/test_data/test_dataset.py:249]: `np.isclose` -> `assert_array_almost_equal`

File: tests/test_accuracy.py [hunk 4]
```diff
@@ -47,6 +49,23 @@ def gt_confusion_matrix(gt_labels, pred_labels, normalize=None):
...
+    detection = ActivityNetDetection(gt_path, result_path)
+
+    results = detection.evaluate()
+    mAP = np.array([
+        0.71428571, 0.71428571, 0.71428571, 0.6875, 0.6875, 0.59722222,
+        0.52083333, 0.52083333, 0.52083333, 0.5
+    ])
+    average_mAP = 0.6177579365079365
+
+    assert np.all(np.isclose(results[0], mAP))
```
Comment (dreamerlin) [inline tests/test_accuracy.py:65]: `assert np.equal` -> `assert_array_equal`
`assert np.isclose` -> `assert_array_almost_equal`

**Comment by kennymckormick** [comment]
@innerlee PR ready for merge

**Review by innerlee — APPROVED** [review]

**Pull request merged**